src/components/ChannelList/ChannelList.tsx

```diff
@@ -194,79 +222,142 @@ ChannelList.propTypes = {
-   *
-   * Default: [ChannelListHeaderErrorIndicator](https://getstream.github.io/stream-chat-react-native/#ChannelListHeaderErrorIndicator)
-   */
-  HeaderErrorIndicator: PropTypes.oneOfType([
-    PropTypes.node,
-    PropTypes.elementType,
-  ]),
-  /**
-   * Custom indicator to display network-down error at top of list, if there is connectivity issue
-   *
-   * Default: [ChannelListHeaderNetworkDownIndicator](https://getstream.github.io/stream-chat-react-native/#ChannelListHeaderNetworkDownIndicator)
-   */
-  HeaderNetworkDownIndicator: PropTypes.oneOfType([
-    PropTypes.node,
-    PropTypes.elementType,
-  ]),
-  /**
-   * Custom UI component to display the list of channels
-   *
-   * Default: [ChannelListMessenger](https://getstream.github.io/stream-chat-react-native/#channellistmessenger)
-   */
-  List: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-  /**
-   * Custom indicator to use when there is error in fetching channels
-   *
-   * Default: [LoadingErrorIndicator](https://getstream.github.io/stream-chat-react-native/#loadingerrorindicator)
-   * */
-  LoadingErrorIndicator: PropTypes.oneOfType([
-    PropTypes.node,
-    PropTypes.elementType,
-  ]),
-  /**
-   * Custom loading indicator to use
-   *
-   * Default: [LoadingIndicator](https://getstream.github.io/stream-chat-react-native/#loadingindicator)
-   * */
-  LoadingIndicator: PropTypes.oneOfType([
-    PropTypes.node,
-    PropTypes.elementType,
-  ]),
-  /**
-   * Custom UI component to display individual channel list items
-   *
-   * Default: [ChannelPreviewMessenger](https://getstream.github.io/stream-chat-react-native/#channelpreviewmessenger)
-   * */
-  Preview: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
+  sort?: ChannelSort<Ch>;
+};
+
+/**
+ * This component fetches a list of channels, allowing you to select the channel you want to open.
+ * The ChannelList doesn't provide any UI for the underlying React Native FlatList. UI is determined by the `List` component which is
+ * provided to the ChannelList component as a prop. By default, the ChannelListMessenger component is used as the list UI.
+ *
+ * @example ../docs/ChannelList.md
+ */
+const ChannelList = <
+  At extends UnknownType = UnknownType,
+  Ch extends UnknownType = UnknownType,
+  Co extends string = LiteralStringForUnion,
+  Ev extends UnknownType = UnknownType,
+  Me extends UnknownType = UnknownType,
+  Re extends UnknownType = UnknownType,
+  Us extends UnknownType = UnknownType
+>(
+  props: PropsWithChildren<Props<At, Ch, Co, Ev, Me, Re, Us>>,
+) => {
+  const {
+    List = ChannelListMessenger,
+    filters = {},
+    lockChannelOrder = false,
+    onAddedToChannel,
+    onChannelDeleted,
+    onChannelHidden,
+    onChannelTruncated,
+    onChannelUpdated,
+    onMessageNew,
+    onRemovedFromChannel,
+    onSelect,
+    options = {},
+    setFlatListRef,
+    sort = {},
```

Comment:
this not something that abc's?

Reply:
this is a holdover from when we were doing caps first, on the others been doing caps in line with the others, so will update this